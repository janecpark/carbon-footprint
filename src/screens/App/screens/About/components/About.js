import React from 'react';

/** About me page shows details of the website */

const About = () => {
  return (
    <div className="About container">
      <h1 className="text-center">About</h1>
      <div className="about-content">
        <div className="about-header">What it is</div>
        <div className="about-text">
          Allows individuals or households in the U.S. to calculate their carbon
          footprint and get personalized advice on managing their climate
          impact.
          <br />
          <br />
          Sign up to save results and compare past results.
        </div>
        <br />
        <hr></hr>
        <div className="about-header">How it is calculated</div>
        <div className="about-text">
          The data is calculated using
          <a href="https://api-central.berkeley.edu/api/11">
            {' '}
            CoolClimate API{' '}
          </a>
          by U.C. Berkeley.
        </div>
        <br />
        <hr></hr>
        <div className="about-header">What information is used</div>
        <div className="about-text">
          User's location, income, household, transportation, public
          transportation, water usage, and food consumption
        </div>
      </div>
    </div>
  );
};

export default About;
