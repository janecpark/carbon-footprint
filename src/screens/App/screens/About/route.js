import { routeConstants } from '../../../../shared/constants';
import About from './';

export default {
  path: routeConstants.ABOUT.route,
  component: About,
};
