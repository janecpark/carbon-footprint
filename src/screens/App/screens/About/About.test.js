import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import About from './components/About';
import {
  UserProvider,
  FormProvider,
  ResultProvider,
  ActionProvider,
} from '../testUtils';

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <About />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <About />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders text on page', () => {
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <About />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  const element = getByText(/About/);
  expect(element).toBeInTheDocument();
});
