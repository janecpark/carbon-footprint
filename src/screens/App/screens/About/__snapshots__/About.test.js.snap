// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="About container"
  >
    <h1
      class="text-center"
    >
      About
    </h1>
    <div
      class="about-content"
    >
      <div
        class="about-header"
      >
        What it is
      </div>
      <div
        class="about-text"
      >
        Allows individuals or households in the U.S. to calculate their carbon footprint and get personalized advice on managing their climate impact.
        <br />
        <br />
        Sign up to save results and compare past results.
      </div>
      <br />
      <hr />
      <div
        class="about-header"
      >
        How it is calculated
      </div>
      <div
        class="about-text"
      >
        The data is calculated using
        <a
          href="https://api-central.berkeley.edu/api/11"
        >
           CoolClimate API 
        </a>
        by U.C. Berkeley.
      </div>
      <br />
      <hr />
      <div
        class="about-header"
      >
        What information is used
      </div>
      <div
        class="about-text"
      >
        User's location, income, household, transportation, public transportation, water usage, and food consumption
      </div>
    </div>
  </div>
</DocumentFragment>
`;
